06-building-a-program-start-to-finish/writing-tests/output/spec/invertedSpec.js

```diff
@@ -0,0 +1,257 @@
+          'invalid. Only array of objects are allowed.'));
+      expect(() => { newIndex.createIndex('test', new Set(['a', 'the'])); })
+          .toThrow(new Error('This file\'s structure is ' +
+          'invalid. Only array of objects are allowed.'));
+      expect(() => { newIndex.createIndex('test', 200); })
+          .toThrow(new Error('This file\'s structure is ' +
+          'invalid. Only array of objects are allowed.'));
+    });
+
+    it('should return error message for files with improper key values', () => {
+      const test = [
+        {
+          tite: 'Alice in Wonderland',
+          text: 'Alice. falls- into@ a+ rabbit hole.',
+        },
+      ];
+
+      expect(() => { newIndex.createIndex('doc1.json', invalidTestFile); })
+        .toThrow(new Error(`Your JSON file does not have the expected format.
+          Documents are to have title and text keys alone.`));
+      expect(() => { newIndex.createIndex('test.json', test); })
+        .toThrow(new Error(`Your JSON file does not have the expected format.
+          Documents are to have title and text keys alone.`));
+    });
+
+    it('should return the correct indices for words in documents', () => {
+      const result = { a: [0],
+        alice: [0],
+        alliance: [1],
+        an: [1],
+        elf: [1],
+        falls: [0],
+        hole: [0],
+        into: [0],
+        man: [1],
+        of: [1],
+        rabbit: [0],
+        unusual: [1] };
+      expect(newIndex.createIndex('doc.json', shortFile2)['doc.json'][0]).toEqual(result);
+    });
+
+    it('returns an object of key file name and value of array of file indices,' +
+     'file name and file titles', () => {
+      const result = [{ a: [0],
+        alice: [0],
+        alliance: [1],
+        an: [1],
+        elf: [1],
+        falls: [0],
+        hole: [0],
+        into: [0],
+        man: [1],
+        of: [1],
+        rabbit: [0],
+        unusual: [1] }, 'doc.json', ['Alice in Wonderland', 'The Lord of the Rings: The Fellowship of the Ring.']];
+      expect(newIndex.createIndex('doc.json', shortFile2)['doc.json']).toEqual(result);
+    });
+
+    it('should only accept an array of objects', () => {
+      const testObj = {
+        title: 'Alice in Wonderland',
+        text: 'Alice. falls- into@ a+ rabbit hole.',
+      };
+
+      expect(() => { newIndex.createIndex('test.json', testObj); })
+       .toThrow(new Error('This file\'s structure is ' +
+       'invalid. Only array of objects are allowed.'));
+    });
+  });
+
+  describe('Tests for the lowerDocText method', () => {
+    it('should turn the text field of documents to lower case', () => {
+      newIndex.lowerDocText(lowerDocFile);
+      lowerDocFile.forEach((doc) => {
+        expect(doc.text).toEqual(doc.text.toLowerCase());
+      });
+    });
+
+    it('only takes an array of objects as argument', () => {
+      expect(() => { newIndex.lowerDocFile({a: 1}) }).toThrowError(TypeError);
```

Comment:
A space is required after '{'. 
A space is required before '}'.

- You will probably want to refactor this entire it block.